client/logic/game/game.gd

```diff
@@ -737,8 +737,7 @@ func _do_state_attacking_clicked_left(local: bool = true) -> void:
 		var damage_result: DamageResult
 		damage_result = _calculate_damage(attacking_unit, defending_unit)
 		defending_unit.last_damage_type = damage_result.weapon_type
-		attacking_unit.last_attack_category = damage_result.weapon_category
-		attacking_unit.play_attack()
+		attacking_unit.play_attack(damage_result.weapon_category)
```

Comment:
I like this!

Reply:
Was thinking the same but the function which uses last_damage_type will be called by an animation (the attacking unit will emit a signal while attacking for every hit which calls that function on the defending unit to show the correct hit effect)